### PR TITLE
*: clean sqlexec in statistics handle

### LINKED ISSUE #32238: consistency is broken since statsReader reads stats parts with different sys sessions
## Enhancement
Now statsReader is using ExecRestrictedSQL to read stats:
https://github.com/pingcap/tidb/blob/master/statistics/handle/handle.go#L1409 

While  ExecRestrictedSQL may get one sys session from sys-session-pool for each sql. It breaks the consistency between different parts of stats, which is the purpose of the design of statsReader, when reading stats without snapshot specified.

### PR DESCRIPTION
**Pull request by chrysan**
### What problem does this PR solve?

Issue Number: close #32238 

Problem Summary:

### What is changed and how it works?
Fix #32238 and btw clean sqlexec calls in handle due to https://github.com/pingcap/tidb/pull/32229 & https://github.com/pingcap/tidb/pull/31618

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]
LGTM, but I think this two pr should be recorded somewhere for the later refactor work. Maybe in the lark document for now, if we don't prepare to file a tracking issue for the new refactor.

**Review by ngaut — APPROVED** [review]
LGTM

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -124,52 +125,16 @@ type Handle struct {
...
 }
 
 func (h *Handle) execRestrictedSQLWithStatsVer(ctx context.Context, statsVer int, sql string, params ...interface{}) ([]chunk.Row, []*ast.ResultField, error) {
-	return h.withRestrictedSQLExecutor(ctx, func(ctx context.Context, exec sqlexec.RestrictedSQLExecutor) ([]chunk.Row, []*ast.ResultField, error) {
-		stmt, err := exec.ParseWithParams(ctx, sql, params...)
-		// TODO: An ugly way to set @@tidb_partition_prune_mode. Need to be improved.
-		if _, ok := stmt.(*ast.AnalyzeTableStmt); ok {
-			pruneMode := h.CurrentPruneMode()
-			if session, ok := exec.(sessionctx.Context); ok {
-				session.GetSessionVars().PartitionPruneMode.Store(string(pruneMode))
```
Comment (xuyifangreeneyes) [inline statistics/handle/handle.go:155]: Here we need to set `sessionVars.PartitionPruneMode` as the latest `@@global.tidb_partition_prune_mode`, which can be done by call `RefreshVars`. Maybe we can use another `OptionFuncAlias` to do that?

Comment (chrysan) [reply statistics/handle/handle.go:155]: Can this logic meet the requirement? https://github.com/pingcap/tidb/blob/master/session/session.go#L1719

**Comment by chrysan** [comment]
> 

OK, will track it.

File: statistics/handle/handle.go (hunk 1 above)
Comment (xuyifangreeneyes) [reply statistics/handle/handle.go:155]: Yes.

**Review by xuyifangreeneyes — APPROVED** [review]

**Comment by hawkingrei** [comment]
@chrysan Please fix the linter.

**Pull request merged**